### PR TITLE
[material-ui] fix: makeSelectable omit inner component's onChange property

### PR DESCRIPTION
**Pull request by ngbrown**
This does update the minimum TypeScript version from 2.8 to 3.5 for the use of `Omit<,>`.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## material-ui-datatables ([<kbd>unpkg</kbd>](https://unpkg.com/browse/material-ui-datatables@latest/))
was missing the following properties:
1. DataTables
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5db8af5c3c457aa907ea5fc1d2e8c4479e47e102
</p>

**Comment by ngbrown** [comment]
The only thing holding this in draft is whether it's worth requiring a newer TypeScript version for this change or not.

**Comment by ngbrown** [comment]
I'll mark this ready to merge in case it's ok to bump the minimum TypeScript version.

**Comment by andrewbranch** [comment]
Makes sense to me, but I’d like to see this approved by another owner before merging.

**Review by FrankBrullo — APPROVED** [review]

**Comment by FrankBrullo** [comment]
It is fine to me too.

**Comment by ngbrown** [comment]
Ready to merge

**Pull request merged**
